factory.go

```diff
@@ -28,21 +38,67 @@ func (f factory) DeploymentType() deployer.DeploymentType {
 	return "python"
 }
 
-func (f factory) ConfigurationSchema() *schema.TypedScopeSchema[*Config] {
+func (f factory) ConfigurationSchema() *schema.TypedScopeSchema[*config.Config] {
 	return Schema
 }
 
-func (f factory) Create(config *Config, logger log.Logger) (deployer.Connector, error) {
+func (f factory) NextConnectorIndex() int {
+	return int(atomic.AddInt64(f.connectorCounter, 1))
+}
+
+func (f factory) Create(config *config.Config, logger log.Logger) (deployer.Connector, error) {
 	pythonPath, err := binaryCheck(config.PythonPath)
 	if err != nil {
-		return &Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
+		return &connector.Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
+	}
+
+	pythonSemver := config.PythonSemVer
+	if pythonSemver == "" {
+		outputSemver, err := f.parsePythonVersion(pythonPath)
+		if err != nil {
+			return nil, err
+		}
+		pythonSemver = *outputSemver
+	}
+
+	connectorFilename := strings.Join([]string{
+		"connector",
+		strings.Replace(pythonSemver, ".", "-", -1),
+		strconv.Itoa(f.NextConnectorIndex())},
+		"_")
+
+	absWorkDir, err := filepath.Abs(config.WorkDir)
+	if err != nil {
+		return nil, fmt.Errorf("error determining absolute path for python deployer's working directory (%w)", err)
+	}
+	connectorFilepath := filepath.Join(absWorkDir, connectorFilename)
+	err = os.MkdirAll(connectorFilepath, os.ModePerm)
```

Comment:
OK, I just always glossed over the `os.ModePerm` definition, but got curious this time and checked. Do we really want to be creating world read-write-execute directories? I'd think we should at least drop write (and possibly execute) for other and perhaps group access??? Actually, I suspect that `os.ModePerm` is intended for use as a mask to extract the permission bits rather than as a value to set.

Like, maybe use 0700 or 0755?

Reply:
That should always be true, but I don't think we should be *asking* for them. That just seems weird.

Reply:
Yeah, but asking for anything else prevents the user from overriding our request (while asking for "wide open" allows the user to get anything they want by setting their `umask` value).

As for weird, it feels that way to me, too, but I blame the interface.  🥴

Reply:
what do you mean by asking for them?

Reply:
`umask` will remove bits you ask for in the mode that the user has masked. For example, my `umask` is `0022` which will remove write from group and other and if I ran this code with a `0777` mask the directory you create should get `0755`.

But if you asked for `0766`, the directory would end up with `0744`, which I think is what I'd tend to prefer for something like this.

However, Webb's not wrong: if you use `0777` as you are, then the user is technically in control of the actual mode by using `umask` and if they don't know about that (or don't use it), that's their problem. I do tend to forget about this bizarre wrinkle in UNIX filesystem semantics, and I've never really liked it, but it's the way it works. (For some definition of "works".)

So, never mind, and I'm just going to resolve this comment. 😆